File: 150.evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        if len(tokens) == 0:
            return 0
        i = 0
        while i < len(tokens) and len(tokens) > 1:
            if tokens[i] == "+":
                tokens[i-2] += tokens[i-1]
            elif tokens[i] == "-":
                tokens[i-2] -= tokens[i-1]
            elif tokens[i] == "*":
                tokens[i-2] *= tokens[i-1]
            elif tokens[i] == "/":
                tokens[i-2] = int(tokens[i-2]/tokens[i-1])
            else:
                tokens[i] = int(tokens[i])
                i += 1
                continue
            i = i-1
            tokens = tokens[:i] + tokens[i+2:]
        return tokens[0]

File: 150.evaluate-reverse-polish-notation.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut s: Vec<i32> = Vec::new();
        for t in tokens.iter() {
            match t.as_str() {
                "+"|"-"|"*"|"/" => {
                    let n1 = s.pop().unwrap();
                    let n2 = s.pop().unwrap();
                    s.push(match t.as_str() {
                        "+" => n1 + n2,
                        "-" => n2 - n1,
                        "*" => n2 * n1,
                        "/" => n2 / n1,
                        _ => 0
                    });
                },
                _ => {
                    s.push(t.parse::<i32>().unwrap());
                }
            }
        }
        s.pop().unwrap()
    }
}
